src/metabase/driver.clj

```diff
@@ -264,6 +264,16 @@
 (defmethod describe-table-fks ::driver [_ _ _]
   nil)
 
+(defmulti describe-table-json
+  "Return information about the nestable columns in a `table`. Required for drivers that support `:nested-field-columns`. Results
+  should match the [[metabase.sync.interface/JSONMetadata]] schema."
+  {:arglists '([this database table])}
```

Comment:
A couple more thoughts:

- You should mention this quickly in https://github.com/metabase/metabase/blob/master/docs/developers-guide/driver-changelog.md so driver authors are aware of it
- Is `describe-table-json` the best name for this, vs. something more generic like `describe-table-nested-columns`? MongoDB documents aren't _really_ JSON, but close enough that I'd be inclined to let it slide; but I can see us supporting some sort of database with a concept of nested columns where the records aren't anything like JSON at all

Reply:
Dumb question:

`describe-table` is already set up to work for nested columns -- this is how MongoDB nested columns are implemented. (See https://github.com/metabase/metabase/blob/master/src/metabase/sync/interface.clj#L34). What's the benefit for creating a new method here vs. just having Postgres use the old method and return stuff under the `:nested-fields` key?

Reply:
Also if we do add a new method, should we rework MongoDB to use it as well and drop `:nested-fields` from `describe-table`? I don't like there being two ways to do one thing

Reply:
The old method is nicely the same for every sql db and is implemented under sql-jdbc. JSON is unfortunately gonna have to be different for every single sql db.

Mongo refactor is a pretty good idea

Reply:
We aren't gonna do XML inasmuch the product peeps didn't even think about XML, looks like, but XML does exist and the implicature if we call it `describe-table-nested-field` is that one day we might to XML or YAML or whatever. I dunno if that's good or bad

Reply:
Dumb question part two: would this method make more sense as a `:sql` or `:sql-jdbc` specific method? If you did that, Mongo could continue to do its own thing the way it currently does (no reason to refactor), but for each SQL DB which needs its own JSON implementation it can implement the SQL driver `describe-table-json` method. Then the SQL or SQL JDBC implementation of `describe-table-fields` can call this method and splice things in and return them in the appropriate format.

If we can it's best to keep the scope of methods as small as absolutely necessary. Otherwise we have to document and explain these methods for people who are writing non-SQL drivers.